programs/nodeos/main.cpp

```diff
@@ -60,9 +63,31 @@ void logging_conf_handler()
 
 void initialize_logging()
 {
+   auto success = fc::log_config::register_appender<dfuse::dm::log_appender>( "deep-mind" );
+   if (!success) {
+      throw std::runtime_error("Unable to correcty register deep mind log appender");
+   }
+
    auto config_path = app().get_logging_conf();
    if(fc::exists(config_path))
      fc::configure_logging(config_path); // intentionally allowing exceptions to escape
+   else {
+      auto cfg = fc::logging_config::default_config();
+      cfg.appenders.push_back(
+         fc::appender_config( "deep-mind", "deep-mind" )
+      );
+
+      fc::logger_config dmlc;
+      dmlc.name = "deep-mind";
+      dmlc.level = log_level::debug;
+      dmlc.enabled = true;
+      dmlc.appenders.push_back("deep-mind");
+
+      cfg.loggers.push_back( dmlc );
+
+      fc::configure_logging(cfg);
```

Comment:
If configured by a `logging.json` then this `else` is not needed.

Reply:
Answering here for https://github.com/EOSIO/eos/pull/8788#discussion_r391109310 also (was unable to comment there for unknown reason).

Ideally, we would have all the settings in the codebase directly, and only `deep-mind = true` would be needed to drive the dmlog, without further manually intervention.

Operation wise, it's easier for user to only activate a single config, without the need to configure a specialized `logging.json` that would enable the logger. 

If you agree, I would enable dmlog "deep-mind" as a valid logger in `fc` submodule that would work out of the box in nodeos, without the need for `logging.json`.

Reply:
Discussion now happening on https://github.com/EOSIO/fc/pull/148